src/processor/graph.monitor.processor.service.ts

```diff
@@ -0,0 +1,137 @@
+import { InjectRedis } from '@liaoliaots/nestjs-redis';
+import { InjectQueue, Processor, WorkerHost } from '@nestjs/bullmq';
+import { Injectable, Logger } from '@nestjs/common';
+import { Job, Queue } from 'bullmq';
+import Redis from 'ioredis';
+import { MILLISECONDS_PER_SECOND } from 'time-constants';
+import { RegistryError } from '@polkadot/types/types';
+import { BlockchainService } from '#app/blockchain/blockchain.service';
+import { BlockchainConstants } from '#app/blockchain/blockchain-constants';
+import { ITxMonitorJob } from '#app/interfaces/monitor.job.interface';
+import { IGraphUpdateJob } from '#app/interfaces/graph-update-job.interface';
+
+@Injectable()
+@Processor('graphTxMonitorQueue')
+export class GraphNotifierService extends WorkerHost {
+  private logger: Logger;
+
+  constructor(
+    @InjectRedis() private cacheManager: Redis,
+    @InjectQueue('graphUpdateQueue') private reconnectionQueue: Queue,
+    private blockchainService: BlockchainService,
+  ) {
+    super();
+    this.logger = new Logger(this.constructor.name);
+  }
+
+  async process(job: Job<ITxMonitorJob, any, string>): Promise<any> {
+    this.logger.log(`Processing job ${job.id} of type ${job.name}`);
+    try {
+      const numberBlocksToParse = BlockchainConstants.NUMBER_BLOCKS_TO_CRAWL;
+      const txCapacityEpoch = job.data.epoch;
+      const previousKnownBlockNumber = (await this.blockchainService.getBlock(job.data.lastFinalizedBlockHash)).block.header.number.toBigInt();
+      const currentFinalizedBlockNumber = await this.blockchainService.getLatestFinalizedBlockNumber();
+      const blockList: bigint[] = [];
+
+      for (let i = previousKnownBlockNumber; i <= currentFinalizedBlockNumber && i < previousKnownBlockNumber + numberBlocksToParse; i += 1n) {
+        blockList.push(i);
+      }
+      const txResult = await this.blockchainService.crawlBlockListForTx(job.data.txHash, blockList, [{ pallet: 'system', event: 'ExtrinsicSuccess' }]);
```

Comment:
Is there some more efficient way we can design this? We may have a lot of txns to monitor, and many of their block ranges will overlap, so we end up scanning the same blocks many times. Seems inefficient...

Reply:
we only look for a tx hash from the block it is  was submitted to to 32n and filter based on tx hash existence in a block

Reply:
Not crucial right now; filed an issue to track: https://github.com/AmplicaLabs/reconnection-service/issues/101